src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -165,6 +165,13 @@ void StorageObjectStorage::Configuration::update(ObjectStoragePtr object_storage
 {
     IObjectStorage::ApplyNewSettingsOptions options{.allow_client_change = !isStaticConfiguration()};
     object_storage_ptr->applyNewSettings(context->getConfigRef(), getTypeName() + ".", context, options);
+    updated = true;
+}
+
+void StorageObjectStorage::Configuration::updateIfRequired(ObjectStoragePtr object_storage_ptr, ContextPtr local_context)
+{
+    if (!updated)
```

Comment:
how critical is it to avoid updating settings multiple time? What happens if it is done concurrently?

Thread sanitizer reports that there is a datarace on `updated` variable:
https://s3.amazonaws.com/altinity-build-artifacts/615/14da835df2d32fe0882cfba308d62a8a8abd0caa/stateless_tests__tsan__[4_4]/stderr.log

So we have two options here:
- convert `updated` to `std::atomic<bool>` so the datarace goes away, but still leaving a chance of concurrent update of the settings.
- add a mutex that is going to guard `Configuration::update`'s body from concurrent updates (+ we  would need some extra checks to avoid updating multiple times)

Reply:
Nothing destructive, but if cache turned off it causes additional requests to remote storage. With cache it changes cache hit rate, also nothing except some tests based on calculation cache hits failed.

Reply:
Yep. I added atomic it in upstream and forget to add for Antalya, it's my fault.